weed/storage/idx/binary_search.go

```diff
@@ -0,0 +1,29 @@
+package idx
+
+import (
+	"github.com/seaweedfs/seaweedfs/weed/storage/types"
+)
+
+// firstInvalidIndex find the first index that do not satisfy the validation function's requirement.
+func FirstInvalidIndex(bytes []byte, validation func(key types.NeedleId, offset types.Offset, size types.Size) (bool, error)) (int, error) {
```

Comment:
can you rename "validation" to "lessThanFn" or "greaterThanFn"?

Reply:
fixed